src/main/java/dev/jbang/JdkManager.java

```diff
@@ -39,7 +39,7 @@ public static Path downloadAndInstallJdk(int version, boolean updateCache) {
 			System.err.println("Installing JDK " + version + "...");
 			UnpackUtil.unpack(jdkPkg, jdkDir);
 			return jdkDir;
-		} catch (IOException e) {
+		} catch (Throwable e) {
```

Comment:
Yes, you're right. This is because before I always rethrew the exception. With this code we'd better split it up: IOExceptions should use this handler while any other Throwables should get rethrown after deleting the folder.